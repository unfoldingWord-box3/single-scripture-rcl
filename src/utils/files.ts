import { createPatch } from 'diff'

/**
 * create a patch of the differences between the contents of originalFile and editedFile
 * @param originalFile
 * @param editedFile
 */
export function getPatch(originalFile, editedFile) {
  const diffResult = createPatch('usfmPatch', originalFile, editedFile)
  return diffResult
}
