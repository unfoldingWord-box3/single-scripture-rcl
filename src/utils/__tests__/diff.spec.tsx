/// <reference types="jest" />
import path from 'path'
import fs from 'fs-extra'
import { getPatch } from '../files'

const fileName = path.join(__dirname, './fixture/57-TIT.usfm')
const titUsfm = fs.readFileSync(fileName, 'utf-8')

describe('diff', () => {
  it('should succeed', () => {
    // given
    const editedFile = titUsfm.replace(`\\v 1 `, `\\v 1 EDITED verse: `)
    // console.log(titUsfm, editedFile)

    // when
    const diffResult = getPatch(titUsfm, editedFile)

    // then
    console.log(diffResult)
    expect(diffResult).toMatchSnapshot()
  })
})
