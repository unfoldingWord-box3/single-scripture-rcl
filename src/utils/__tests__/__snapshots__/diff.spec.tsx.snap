// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`diff should succeed 1`] = `
"Index: usfmPatch
===================================================================
--- usfmPatch
+++ usfmPatch
@@ -9,9 +9,9 @@
 
 \\ts\\*
 \\c 1
 \\p
-\\v 1 \\zaln-s |x-strong="G39720" x-lemma="Παῦλος" x-morph="Gr,N,,,,,NMS," x-occurrence="1" x-occurrences="1" x-content="Παῦλος"\\*\\w Paul|x-occurrence="1" x-occurrences="1"\\w*\\zaln-e\\*,
+\\v 1 EDITED verse: \\zaln-s |x-strong="G39720" x-lemma="Παῦλος" x-morph="Gr,N,,,,,NMS," x-occurrence="1" x-occurrences="1" x-content="Παῦλος"\\*\\w Paul|x-occurrence="1" x-occurrences="1"\\w*\\zaln-e\\*,
 \\zaln-s |x-strong="G14010" x-lemma="δοῦλος" x-morph="Gr,N,,,,,NMS," x-occurrence="1" x-occurrences="1" x-content="δοῦλος"\\*\\w a|x-occurrence="1" x-occurrences="1"\\w*
 \\w servant|x-occurrence="1" x-occurrences="1"\\w*\\zaln-e\\*
 \\zaln-s |x-strong="G23160" x-lemma="θεός" x-morph="Gr,N,,,,,GMS," x-occurrence="1" x-occurrences="2" x-content="Θεοῦ"\\*\\w of|x-occurrence="1" x-occurrences="5"\\w*
 \\w God|x-occurrence="1" x-occurrences="2"\\w*\\zaln-e\\*
"
`;
